### PR TITLE
BKNDLSS-33608 [Mapbox] (Support) Add the possibility to make markers active programatically.

### PR DESCRIPTION
**Pull request by addKiwi**

### CODE REVIEW

File: components/bl-mapbox/src/helpers.js [hunk 1]
```diff
@@ -224,8 +235,14 @@ const removeMarkers = markers => {
...
-const changeMarkerColor = marker => {
-  marker.getElement().classList.toggle('marker-root--active');
+export const changeMarkerColor = (marker, status) => {
+  const element = marker?.getElement();
+
+  if (element && status) {
+    element.classList.add('marker-root--active');
+  } else if (element) {
+    element.classList.remove('marker-root--active');
+  }
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:245]:
```suggestion
  if (element ) {
    element?.classList.toggle('marker-root--active', status);
  }
```

File: components/bl-mapbox/src/helpers.js [hunk 2]
```diff
@@ -176,12 +176,16 @@ export const initMapboxLibrary = (mapRef, mapContainerRef, component, eventHandl
 
 };
 
-export const useMarkers = (markers, mapRef, onMarkerClick) => {
+export const useMarkers = (markers, mapRef, eventHandlers) => {
+  const { onMarkerClick, onMarkersCreated } = eventHandlers;
   const markerElements = useRef();
+  const collectedMarkerData = useRef([]);
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:182]: Why do we need ref for that? Could it be just local variable?

File: components/bl-mapbox/component.json [hunk 3]
```diff
@@ -417,6 +434,20 @@
...
+    {
+      "id": "setMarkerStatus",
+      "label": "Set Marker Status",
+      "inputs": [
+        {
+          "id": "marker",
+          "label": "Marker"
+        },
+        {
+          "id": "status",
```
Comment (v-excelsior) [inline components/bl-mapbox/component.json:447]: `Status" true` at screenshots looks not right. Is should be `status: 'active'` or `active: true`

File: components/bl-mapbox/component.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "id": "c_8f971f6aef45e2a67d942f97229a378a",
```
Comment (v-excelsior) [inline components/bl-mapbox/component.json:2]: Is it OK that id was changed?

File: components/bl-mapbox/component.json [hunk 5]
```diff
@@ -208,8 +208,13 @@
         {
           "id": "data",
           "label": "Data"
+        },
+        {
+          "id": "marker",
```
Comment (v-excelsior) [inline components/bl-mapbox/component.json:213]: Where I could see Market signature?

File: components/bl-mapbox/component.json [hunk 6]
```diff
@@ -417,6 +434,20 @@
...
+      "id": "setMarkerStatus",
+      "label": "Set Marker Status of",
+      "inputs": [
+        {
+          "id": "marker",
+          "label": "Marker"
+        },
+        {
+          "id": "active",
+          "label": "Active"
```
Comment (v-excelsior) [inline components/bl-mapbox/component.json:448]: Status = Marker?

Opposite to **Active** is **Inactive**, but why "Marker"?

File: components/bl-mapbox/src/helpers.js [hunk 7]
```diff
@@ -224,10 +243,14 @@ const removeMarkers = markers => {
   }
 };
 
-const changeMarkerColor = marker => {
-  marker.getElement().classList.toggle('marker-root--active');
+export const changeMarkerColor = (marker, status) => {
+  const element = marker?.getElement();
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:247]: empty line

File: components/bl-mapbox/src/actions.js [hunk 8]
```diff
@@ -1,22 +1,25 @@
...
+    loaded         : () => map?.loaded(),
+    getCenter      : () => map?.getCenter(),
+    setCenter      : (center, eventData) => map?.setCenter(center, eventData),
+    panBy          : (offset, options, eventData) => map?.panBy(offset, options, eventData),
+    panTo          : (lnglat, options, eventData) => map?.panTo(lnglat, options, eventData),
+    getZoom        : () => map?.getZoom(),
+    setZoom        : (zoom, eventData) => map?.setZoom(zoom, eventData),
+    flyTo          : (options, eventData) => map?.flyTo(options, eventData),
+    rotateTo       : (bearing, options, eventData) => map?.rotateTo(bearing, options, eventData),
+    setMarkerStatus: (marker, active) => changeMarkerColor(marker, active),
```
Comment (v-excelsior) [inline components/bl-mapbox/src/actions.js:23]: changeMarkerColor -> setMarkerActive
setMarkerStatus -> setMarkerActive

In docs also: set marker status -> set marker active

File: components/bl-mapbox/src/helpers.js [hunk 9]
```diff
@@ -126,6 +126,14 @@ const appendDirectionsOnMobile = (directions, mapRef, mapContainerRef) => {
   mapContainerRef.current.parentNode.insertAdjacentElement('afterend', div);
 };
 
+export const isMapWidthChanged = (mapRef, isMapLoaded) => {
+  useEffect(() => {
+    if (isMapLoaded) {
+      console.log('width changed to ', mapRef?.current?.transform?.width);
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:132]: Why do we need this log?

File: components/bl-mapbox/src/helpers.js [hunk 10]
```diff
@@ -176,12 +184,16 @@ export const initMapboxLibrary = (mapRef, mapContainerRef, component, eventHandl
...
-export const useMarkers = (markers, mapRef, onMarkerClick) => {
+export const useMarkers = (markers, mapRef, eventHandlers) => {
+  const { onMarkerClick, onMarkersCreated } = eventHandlers;
   const markerElements = useRef();
+  const collectedMarkerData = [];
 
   useEffect(() => {
     removeMarkers(markerElements.current);
 
+    collectedMarkerData.length = 0;
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:195]: 0_0

Comment (v-excelsior) [reply components/bl-mapbox/src/helpers.js:195]: Maybe we should create `collectedMarkerData` inside useEffect?

File: components/bl-mapbox/src/helpers.js [hunk 11]
```diff
@@ -224,10 +243,14 @@ const removeMarkers = markers => {
   }
 };
 
-const changeMarkerColor = marker => {
-  marker.getElement().classList.toggle('marker-root--active');
+export const changeMarkerColor = (marker, status) => {
+  const element = marker?.getElement();
+  if (element) {
+    element?.classList.toggle('marker-root--active', status);
```
Comment (v-excelsior) [inline components/bl-mapbox/src/helpers.js:249]: Why do we need `?.` if already check for `element` in if condition?

File: components/bl-mapbox/README.md [hunk 12]
```diff
@@ -98,3 +100,47 @@ Addition of polygons on map:
...
+
+````
+<block xmlns="http://www.w3.org/1999/xhtml" type="set_object_property" id="|*/tH(rPYe?30*/VaoGB" x="418" y="168"><value name="object"><block type="root_block_ui_builder_common__context_blocks_pageData" id="-=]V$$gcXx]2nQuoSJe0" bl_meta="{&quot;label&quot;:&quot;Page Data&quot;}"></block></value><value name="propName"><shadow type="text" id="rz,3/rLr%;rCie9*nqq_"><field name="TEXT">createdMarkers</field></shadow></value><value name="propValue"><block type="root_block_ui_builder_c_4d14629c33bdf5b6b50ffd72e25e5ca2_onMarkersCreated_handler_context_blocks_markers" id="?;I7vs*JOYgHLedHBf#-" bl_meta="{&quot;label&quot;:&quot;Markers&quot;}"></block></value></block>
+````
+
+</details>
+
+3. Find and get the required marker and highlight it.
+
+![highlight markers example](./example-images/mapbox-markers-highlight-marker.png)
```
Comment (v-excelsior) [inline components/bl-mapbox/README.md:132]: Screenshots also should be updated due to changes

**Review by v-excelsior — CHANGES_REQUESTED** [review]

File: components/bl-mapbox/component.json (hunk 4 above)
Comment (addKiwi) [reply components/bl-mapbox/component.json:2]: Just created a copy of the component to separate tickets. Could it be a problem after republishing?

File: components/bl-mapbox/component.json (hunk 5 above)
Comment (addKiwi) [reply components/bl-mapbox/component.json:213]: I think we don't need it, I don't see cases where we use the 'marker'. 'coordinates', 'description' and 'data' enough.

File: components/bl-mapbox/component.json (hunk 4 above)
Comment (v-excelsior) [reply components/bl-mapbox/component.json]: You should ask Stas, but I think yes

**Pull request merged**